### DISCUSSION TITLE
0.4.0-alpha

### DISCUSSION THREAD

**Discussion by toerob** [toerob/vscode-tads3tools#29] [Announcements]
Added

- A command to add files and create makefile entries has been added
- Macro definitions are now also indexed and can be peeked upon and opened up with CTRL-click.

Fixed

- Fixed an issue with definitions of propertyset that weren't displayed correctly
- Fixed issue with the errors parser so it now displays the correct source file of the error
- Fixed issue so that all symbols are found with the workspace symbol search (CTRL-T)
- Fixed issue that was caused by auto closing quote pairs within strings

Many thanks to John Ziegler for the extensive bug reporting!

Notes

If earlier versions has been used the library cache might need be cleared, or you will still suffer from numerical artefacts in the outliner on propertyset symbols. 

1. Run the command with ctrl-shift-P: "Tads3 command: Clear cache for standard libraries adv3/adv3Lite."
2. Either: restart vscode, or run the following command with ctrl-shift-P: "Tads3 command: Set which makefile (.t3m)..." (this will trigger a complete reparse)

<hr /><em>This discussion was created from the release <a href='https://github.com/toerob/vscode-tads3tools/releases/tag/v0.4.0-alpha'>0.4.0-alpha</a>.</em>

**Comment by johnnywz00** [comment]
Thanks for adding all the workspace symbols! Is it my imagination, or did I read in one of your posts that it is possible to exclude the library symbols from the finder?

**Reply by toerob** [reply]
Nothing that _filters_ the result for the workspace symbol provider but if the setting “tads3.parseOnlyTheWorkspaceFiles“ is true and you don’t have any cached library files you won’t see them either. But I’m guessing that’s not what you wish for here?

**Reply by johnnywz00** [reply]
Okay, that's probably what I read... I like having the library symbols accessible for peek definitions, but within the workspace symbol finder there are so many symbols (and I'm usually only using that feature to hop to my own objects) that it takes a couple of seconds to load the options every time I type something.  My game is a large one, I don't know if that's the difference; but if it was easy to have an option of only searching through my own symbols, it would be nice... don't pressure yourself about it though!

**Reply by toerob** [reply]
No problems! A filter setting might be good to add.  I also wonder if some pure caching in addition to filtering might help out here. I noticed a small delay when implementing it but since it was just about a second for a small project I didn’t worry about optimizing it.
